File: app/models/result.rb
class Result
  include SoundcloudHelper
  attr_reader :soundcloud_id, :user_name, :title, :duration, :stream_url

  def initialize(track_data)
    @soundcloud_id = track_data.fetch(:id, 0)
    @user_name = track_data[:user].fetch(:username, 'username')
    @title = track_data.fetch(:title, 'title')
    @duration = track_data.fetch(:duration, 0)
    @stream_url = track_data.fetch(:stream_url, '/') + "?client_id=#{ENV['FLOWLIST_SOUNDCLOUD_CLIENT_ID']}"
  end
end

File: app/assets/javascripts/models/result.js
function Result(id, data) {
  this.id           = id + 1;
  this.soundcloudId = data.soundcloud_id;
  this.userName     = data.user_name;
  this.title        = data.title;
  this.duration     = data.duration;
  this.streamUrl    = data.stream_url;
}

Result.prototype = {
  init: function(){
    $('body').append(tracksTemplate(this));
    return this;
  },
  underscored: function(){
    var mappedSong = {};
    for (var key in this) {
      mappedSong[key.toUnderscore()] = this[key];
    }
    return mappedSong;
  }
};